dataRetriever/factory/dataPoolFactory.go

```diff
@@ -83,10 +90,37 @@ func NewDataPoolFromConfig(args ArgsDataPool) (dataRetriever.PoolsHolder, error)
 		return nil, err
 	}
 
-	cacherCfg = factory.GetCacherFromConfig(mainConfig.TrieNodesDataPool)
-	trieNodes, err := storageUnit.NewCache(cacherCfg)
+	cacher, err := capacity.NewCapacityLRU(
+		int(mainConfig.TrieSyncStorage.Capacity),
+		int64(mainConfig.TrieSyncStorage.SizeInBytes),
+	)
+
+	dbCfg := factory.GetDBFromConfig(mainConfig.TrieSyncStorage.DB)
+	argDB := storageUnit.ArgDB{
+		DBType:            dbCfg.Type,
+		Path:              dbCfg.FilePath,
+		BatchDelaySeconds: dbCfg.BatchDelaySeconds,
+		MaxBatchSize:      dbCfg.MaxBatchSize,
+		MaxOpenFiles:      dbCfg.MaxOpenFiles,
+	}
+
+	if mainConfig.TrieSyncStorage.UseTmpAsFilePath {
+		filePath, err := ioutil.TempDir("", "trieSyncStorage")
```

Comment:
I do not think it will be helpful, because this is only one function call. Or do you mean the whole section, something like `GetDbWithTempPath(cfg storageUnit.DBConfig) (storage.Persister,error)` ?

Reply:
can remaind like this